xdsl/ir.py

```diff
@@ -128,7 +128,7 @@ class SSAValue(ABC):
     _name: str | None = field(init=False, default=None)
 
     _name_regex: ClassVar[re.Pattern[str]] = re.compile(
-        r'[A-Za-z0-9._$-]*[A-Za-z._$-]')
+        r'([0-9]+|([A-Za-z_$.-][\w$.-]*))')
```

Comment:
Should we actually keey the pure `[0-9+]` case?
Since we don't support this case during parsing, should we remove it here? I feel this would make our project more consistent.

Reply:
I am not sure about this, I did not remove them before because they are valid SSA value names. But I have no strong opinions either way. I can see that removing these might improve the readability of the printed code in cases where you accidentally assign pure numbers to names, but I can also see some users confused that they can't set a pure number string as name.

I'm fine with both, we may want to see what mlir is doing in this case.

Reply:
Well, so MLIR won't save names if they are only numbers.
What I mean here, is that `SSAValue` can have integer names, and when I mean names I mean saved names. But the printer won't ever save an integer name. Which is why I think we should change the name case for `SSAValue`, since it won't ever be used.